src/Raven.Studio/typescript/viewmodels/resources/serverDashboard.ts

```diff
@@ -668,13 +670,24 @@ class driveUsageSection {
         });
 
         items.forEach(incomingItem => {
-            const matched = this.table().find(x => x.mountPoint() === incomingItem.MountPoint);
+            const incomingItemMountPoint = incomingItem.MountPoint;
+            const matched = this.table().find(x => x.mountPoint() === incomingItemMountPoint);
+            
             if (matched) {
                 matched.update(incomingItem);
             } else {
-                const usage = new driveUsage(incomingItem, this.storageChart.getColorClassProvider(), this.includeTemporaryBuffers);
+                const usage = new driveUsage(incomingItem, this.includeTemporaryBuffers, this.storageChart);
                 this.table.push(usage);
             }
+            
+            if (!this.numberOfDatabasesPerMountPoint[incomingItemMountPoint]) {
+                this.numberOfDatabasesPerMountPoint[incomingItemMountPoint] = 0;
+            }
+            
+            if (incomingItem.Items.length !== this.numberOfDatabasesPerMountPoint[incomingItemMountPoint]) {
+                this.numberOfDatabasesPerMountPoint[incomingItemMountPoint] = incomingItem.Items.length;
+                this.graphData();
```

Comment:
please keep in mind that you are calling graphData in loop, as result it will redraw graph multiple times. Can we avoid doing that?

Reply:
so basically we can redraw if *any* count has mismatch.

Reply:
right - fixed